src/logic/sourcemap.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable no-bitwise */
+
+// Adapted from https://github.com/Rich-Harris/vlq, thanks!
```

Comment:
Why not use https://www.npmjs.com/package/vlq or https://www.npmjs.com/package/sourcemap-codec (I've not looked into their differences) as a dependency?

Reply:
I didn't want to introduce another dependency but I will just use that if its ok?